src/prototype_rust_sdk/workflow_future.rs

```diff
@@ -255,12 +296,6 @@ impl Future for WorkflowFuture {
                 }
             }
 
-            if activation_cmds.is_empty() {
-                panic!(
-                    "Workflow did not produce any commands or exit, but awaited. This \
-                 means it will deadlock. You probably awaited on a non-WfContext future."
-                );
-            }
```

Comment:
Ah, this lines up with the weird non-activations. I'd really like to keep this unless there's a really good reason it needs to go.

Reply:
Same response as above

Reply:
Yeah so this can't stay as-is then, clearly. It's still a nice thing to be able to detect, though. Maybe we can upgrade this to also check that there is nothing that needs to be unblocked. *then* it's a deadlock. If that's easy go for it, if not we can fix in another PR